pkg/kubelet/dockertools/docker.go

```diff
@@ -691,3 +693,63 @@ type ContainerCommandRunner interface {
 	RunInContainer(containerID string, cmd []string) ([]byte, error)
 	GetDockerServerVersion() ([]uint, error)
 }
+
+type DockerCache interface {
+	KubeletRunningDockerContainers() (DockerContainers, error)
+}
+
+func NewDockerCache(client DockerInterface) DockerCache {
+	containers, err := GetKubeletDockerContainers(client, false)
+	cacheTime := time.Now()
+	return &dockerCache{
+		client:        client,
+		cacheTime:     cacheTime,
+		containers:    containers,
+		err:           err,
+		updatingCache: false,
+	}
+}
+
+// dockerCache is a default implementation of DockerCache interface
+type dockerCache struct {
+	client        DockerInterface
```

Comment:
Can we document these fields a little? Some are obvious but others aren't without looking at the implementation :)